framework/src/problems/FEProblemBase.C

```diff
@@ -7900,6 +7900,11 @@ FEProblemBase::adaptMesh()
     }
     else
     {
+      // If the mesh didn't change, we still need to update the displaced mesh
+      // to undo the undisplacement performed in Adaptivity::adaptMesh
+      if (_displaced_problem)
+        _displaced_problem->updateMesh();
```

Comment:
Did you try putting this in `Adaptivity::adaptMesh` and removing the call to `updateMesh()` in `DisplacedProblem::meshChanged()`? If that didn't break anything, I think that would be my preferred change. A developer who looks only at this code and the lines immediately surrounding it will wonder if/where the re-displacement happens when `mesh_changed_this_step`. It would also conceptually make good sense to undisplace and re-displace in the same method

Reply:
I tried that at first, but the solutions need to be projected first, which I believe happens in `meshChangedHelper`. I can add it it in `Adaptivity::adaptMesh` for the case when no refinement happens, but the call in `meshChangedHelper()` would need to remain.

Reply:
Ok that's fine. Let's make sure what you just told me is included in a comment somewhere in the code though

Reply:
Sounds good. I added an explanation in the doxygen of Adaptivity::adaptMesh.